src/server/frontend/userstate.js

```diff
@@ -0,0 +1,57 @@
+import config from '../config';
+import Promise from 'bluebird';
+import Immutable from 'immutable';
+
+// Example how initialState, which is the same for all users, is enriched with
+// user state. With state-less Flux, we don't need instances.
+export default function userState() {
+
+  return (req, res, next) => {
+    loadUserData(req).then(loadedData => {
+      req.userState = Immutable.Map().mergeWith(...loadedData);
```

Comment:
I believe `mergeWith` is used wrongly in this case. The signature is as follows:
```
mergeWith(
merger: (previous?: V, next?: V, key?: K) => V,
...iterables: Iterable<K, V>[]
): Map<K, V>
```
I think `merge` will work, also - no need to destruct.